drake/util/drakeGradientUtil.h

```diff
@@ -1,435 +1,6 @@
-      auto_diff_matrix.size(), num_variables);
-  for (int row = 0; row < auto_diff_matrix.rows(); row++) {
-    for (int col = 0; col < auto_diff_matrix.cols(); col++) {
-      auto gradient_row =
-          gradient.row(row + col * auto_diff_matrix.rows()).transpose();
-      if (auto_diff_matrix(row, col).derivatives().size() == 0) {
-        gradient_row.setZero();
-      } else {
-        gradient_row = auto_diff_matrix(row, col).derivatives();
-      }
-    }
-  }
-  return gradient;
-}
-
-template <typename DerivedGradient, typename DerivedAutoDiff>
-void gradientMatrixToAutoDiff(
-    const Eigen::MatrixBase<DerivedGradient>& gradient,
-    Eigen::MatrixBase<DerivedAutoDiff>& auto_diff_matrix) {
-  typedef typename Eigen::MatrixBase<DerivedGradient>::Index Index;
-  auto nx = gradient.cols();
-  for (Index row = 0; row < auto_diff_matrix.rows(); row++) {
-    for (Index col = 0; col < auto_diff_matrix.cols(); col++) {
-      auto_diff_matrix(row, col).derivatives().resize(nx, 1);
-      auto_diff_matrix(row, col).derivatives() =
-          gradient.row(row + col * auto_diff_matrix.rows()).transpose();
-    }
-  }
-}
-
-namespace Drake {
-namespace internal {
-template <typename Derived, typename Scalar>
-struct ResizeDerivativesToMatchScalarImpl {
-  static void run(Eigen::MatrixBase<Derived>& mat, const Scalar& scalar){}
-};
-
-template <typename Derived, typename DerivType>
-struct ResizeDerivativesToMatchScalarImpl<Derived,
-                                          Eigen::AutoDiffScalar<DerivType> > {
-  using Scalar = Eigen::AutoDiffScalar<DerivType>;
-  static void run(Eigen::MatrixBase<Derived>& mat, const Scalar& scalar) {
-    for (int i = 0; i < mat.size(); i++) {
-      auto& derivs = mat(i).derivatives();
-      if (derivs.size() == 0) {
-        derivs.resize(scalar.derivatives().size());
-        derivs.setZero();
-      }
-    }
-  }
-};
-}
-
-/** Resize derivatives vector of each element of a matrix to to match the size
- * of the derivatives vector of a given scalar.
- * \brief If the mat and scalar inputs are AutoDiffScalars, resize the
- * derivatives vector of each element of the matrix mat to match
- * the number of derivatives of the scalar. This is useful in functions that
- * return matrices that do not depend on an AutoDiffScalar
- * argument (e.g. a function with a constant output), while it is desired that
- * information about the number of derivatives is preserved.
- * \param mat matrix, for which the derivative vectors of the elements will be
- * resized
- * \param scalar scalar to match the derivative size vector against.
- */
-template <typename Derived>
-void resizeDerivativesToMatchScalar(Eigen::MatrixBase<Derived>& mat,
-                                    const typename Derived::Scalar& scalar) {
-  internal::ResizeDerivativesToMatchScalarImpl<
-      Derived, typename Derived::Scalar>::run(mat, scalar);
-}
-}
+#include "drake/core/drakeGradientUtil.h"
```

Comment:
Perhaps this file should also have a deprecation warning via '#pragma message' or similar.